site-scrapers/EastBostonNHC/index.js

```diff
@@ -0,0 +1,123 @@
+        return {
+            name: `${site.name} (${loc.city} - ${loc.zip})`,
+            hasAvailability: !!response,
+            availability: response?.availability || {},
+            signUpLink: site.website,
+            extraData: `Open to residents of the following neighborhoods: Chelsea (02150), East Boston (02128), Everett (02149), Revere (02151), South End (02118), Winthrop (02152)`,
+            ...loc,
+            timestamp: moment().format(),
+        };
+    });
+};
+
+async function GetAllAvailability(availabilityService) {
+    const accessToken = (await availabilityService.getLoginResponse()).token;
+    const rawAvailability = await availabilityService.getAvailabilityResponse(
+        accessToken
+    );
+    return rawAvailability.response.reduce((acc, appointment) => {
+        const zip = appointment.facility.postcode;
+        const date = appointment.date.split("T")[0]; // get 2021-03-08 from 2021-03-08T22:40:00.000Z
+        if (!acc[zip]) {
+            // in case it's our first time seeing this zip...
+            return {
+                [zip]: {
+                    availability: {
+                        [date]: {
+                            hasAvailability: true,
+                            numberAvailableAppointments: 1,
+                        },
+                    },
+                },
+                ...acc,
+            };
+        }
+        if (!acc[zip].availability[date]) {
+            // in case we've seen this zip before, but not this date...
+            return lodash.set(acc, `${zip}.availability.${date}`, {
+                hasAvailability: true,
+                numberAvailableAppointments: 1,
+            });
+        }
+        // otherwise, we increment b/c we've seen this zip and date before
+        return lodash.set(
+            acc,
+            `${zip}.availability.${date}.numberAvailableAppointments`,
+            acc[zip].availability[date].numberAvailableAppointments + 1
+        );
+    }, {});
+}
+
+async function getHttpsResponse(url, method, headers, inputBody) {
+    const responsePromise = new Promise((resolve) => {
+        const req = https.request(
+            url,
+            {
+                method,
+                headers,
+            },
+            (res) => {
+                let body = "";
+                res.on("data", (chunk) => {
+                    body += chunk;
+                });
+                res.on("end", () => {
+                    resolve(body);
+                });
+            }
+        );
+        req.on("error", (error) => {
+            console.error(error);
+        });
+        if (inputBody) {
+            req.write(inputBody);
+        }
+        req.end();
+    });
+    return JSON.parse(await responsePromise);
+}
+
+function getAvailabilityService() {
+    return {
+        async getLoginResponse() {
+            const data = JSON.stringify({ id: "600f45213901d90012deb171" });
+            return await getHttpsResponse(
```

Comment:
I've done it before as well, but we should probably use `node-fetch` instead of rolling our own Promise wrapper to `https` every time. This is going to go wrong in some subtle way eventually and will be hell to debug.

Reply:
changed to use node-fetch instead.
I think the previous way was OK but `node-fetch` is less verbose and I took it as an opp to learn it